walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,159 @@
+                                available.add(TimeRange.fromStartDuration(optional.start(), optional.duration()));
+                            } else {
+                                if(!available.contains(TimeRange.fromStartDuration(mandatory.start(), mandatory.duration()))){
+                                    available.add(TimeRange.fromStartDuration(mandatory.start(), mandatory.duration()));
+                                }
+                            }
+                        }
+                        if(mandatory.overlaps(optional) && request.getDuration() <= (Math.min(optional.end(), mandatory.end())- Math.max(optional.start(), mandatory.start())-1)){
+                            if(!available.contains(TimeRange.fromStartEnd(Math.max(optional.start(), mandatory.start()), Math.min(optional.end(), mandatory.end()), false))){
+                                available.add(TimeRange.fromStartEnd(Math.max(optional.start(), mandatory.start()), Math.min(optional.end(), mandatory.end()), false));
+                            }
+                        }
+                    }
+                }
+                if(available.isEmpty()){
+                    return availableMandatory;
+                } else {
+                    return available;
+                }
+
+            }
+        }
+    }
+
+    private ArrayList<TimeRange> getAvailability(ArrayList<TimeRange> busyTimes, MeetingRequest request){
+        ArrayList<TimeRange> availableTimes = new ArrayList<>();
+
+        if(request.getDuration() > TimeRange.END_OF_DAY){
+            return availableTimes;
+        }
+        
+        if(busyTimes.isEmpty()){
+            availableTimes.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true));
+            return availableTimes;
+        }
+
+        if(busyTimes.size()==1){
+            if(busyTimes.get(0) == TimeRange.WHOLE_DAY){
+                return availableTimes;
+            } else {
+                availableTimes.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, busyTimes.get(0).start(), false));
+                availableTimes.add(TimeRange.fromStartEnd(busyTimes.get(busyTimes.size()-1).end(), TimeRange.END_OF_DAY, true));
+                return availableTimes;
+            }
+        }
+
+        if(busyTimes.get(0).start()!= 0){
+            availableTimes.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, busyTimes.get(0).start(), false));
+        }
+
+        //adds possible time range for the avaiable times when none of the groups meetings overlap and ensures that the duration in available  times is more or equivalent to requested duration
+        for( int timeIndex = 1; timeIndex < busyTimes.size(); timeIndex++){
+            if(!busyTimes.get(timeIndex).overlaps(busyTimes.get(timeIndex-1)) && request.getDuration()<=(busyTimes.get(timeIndex).start()) - busyTimes.get(timeIndex-1).end()){
```

Comment:
and also where did you handle the overlap cases?

Reply:
I don't. If the time overlap, i continue the loop and don't do anything because there wont be any available time between those two time frames. The next iteration will just check the past current time frame and the new current.